python/ray/serve/_private/http_util.py

```diff
@@ -217,14 +217,27 @@ class ASGIReceiveProxy:
 
     def __init__(
         self,
+        scope: Scope,
         request_id: str,
         receive_asgi_messages: Callable[[str], Awaitable[bytes]],
     ):
+        self._type = scope["type"]  # Either 'http' or 'websocket'.
         self._queue = asyncio.Queue()
         self._request_id = request_id
         self._receive_asgi_messages = receive_asgi_messages
         self._disconnect_message = None
 
+    def _get_default_disconnect_message(self) -> Message:
+        """Return the appropriate disconnect message based on the connection type."""
```

Comment:
Nitpick (non-blocker): Would be nice to also link the specs here https://asgi.readthedocs.io/en/latest/specs/www.html#disconnect-receive-event-ws and https://asgi.readthedocs.io/en/latest/specs/www.html#disconnect-receive-event

Reply:
done